### PR TITLE
Validator start

### PR DESCRIPTION
**Pull request by kengie**
Start of prototyping validation tool, using Pandera. Schemas folder has the data frame schemas, we are making one per form. The 2 notebooks contain the code for cleaning input test data and testing out schema rules. 

The `schemas` folder has all of the schemas so far, and the `data` folder has the data we are using (public from NTD). 

Also updated the Readme.

### CODE REVIEW

File: validation_tool/schemas/facilities_a10_schema.py [hunk 1]
```diff
@@ -0,0 +1,210 @@
+from pandera import DataFrameSchema, Column, Check, Index, MultiIndex
```
Comment (hunterowens) [inline validation_tool/schemas/facilities_a10_schema.py]: add description in comments at top of why / what schema is.

File: validation_tool/schemas/facilities_a15_schema.py [hunk 2]
```diff
@@ -0,0 +1,357 @@
+from pandas import Timestamp
```
Comment (hunterowens) [inline validation_tool/schemas/facilities_a15_schema.py]: same as above. what schema is that, what does it map to

File: validation_tool/pandera_test.ipynb
Comment (hunterowens) [inline validation_tool/pandera_test.ipynb:1]: lets break this out into a script.

File: validation_tool/schemas/a10_inferred_schema.py [hunk 3]
```diff
@@ -0,0 +1,597 @@
+rom pandera import DataFrameSchema, Column, Check, Index, MultiIndex
```
Comment (hunterowens) [inline validation_tool/schemas/a10_inferred_schema.py]: syntaxerror, missing `f`. should read `from pandera`

File: README.md [hunk 4]
```diff
@@ -1 +1,10 @@
...
+# ntd-modernization
+  
+This repo contains analyses and code related to helping make Caltrans NTD reporting more efficient. 
+  
+### Issues Analysis, 2020 - 2022
+The `ntd_issues_analysis.Rmd` R Markdown file analyses errors Caltrans received from NTD from 2020 - 22. The accompanying HTML file shows code and web-interactive graphics (if it does not render, download onto your computer and reopen).  
+  
+---
+### Pre-submission Validator
+The folder `validator_tool` contains code for prototyping a presubmission validation tool to anticipate issues in submitting NTD reports, based on 2020-22 issues.
```
Comment (hunterowens) [inline README.md]: missing `requirements.txt` or other software definition package. also missing  how to run.

File: validation_tool/pandera_test.ipynb
Comment (hunterowens) [inline validation_tool/pandera_test.ipynb:1]: all files / notebooks should have purpose and how run on the start.

**Review by hunterowens — CHANGES_REQUESTED** [review]
Mostly good and glad to see the validator starting. Needs significant work on project scaffolding and making it runnable with clear instruction. 

Would like to see it possible for @csuyat-dot be able to run this in the notebook environment before we merge it in.

File: validation_tool/schemas/facilities_a10_schema.py
Comment (KatrinaMKaiser) [inline validation_tool/schemas/facilities_a10_schema.py:1]: Remove spaces from column names

File: README.md (hunk 4 above)
Comment (kengie) [reply README.md]: This is not a library or executable repo as-is, so the readme is more a description of what is in each folder, with some showing intermediate development work. I think as the infrastructure gets put in place, we can chunk out code sections that are executable - and in sub-areas describe how to run things?

**Comment by kengie** [comment]
> Mostly good and glad to see the validator starting. Needs significant work on project scaffolding and making it runnable with clear instruction.
> 
> Would like to see it possible for @csuyat-dot be able to run this in the notebook environment before we merge it in.

I significantly changed this code and reorganized its folders. There are now 2 executable python files - instructions on how to run each are inside the files at the top. They each contain a set of validations on a certain NTD form or natural grouping of data questions. The readme is updated to reflect the organization and what is where. Inputs are data tables and outputs are excel files for the Caltrans team to use to follow up with agencies. This will change over time as the API is built and as we put this on the cloud.

Checking in on our repo objectives and workflow  - this code is WIP and not really what we will have ready to deploy even as a prototype yet. All good if we don't want to merge in intermediate work - I can make commits as I go, but not make a PR until something full is ready to go? 

I have decided not to use Pandera yet - with each set of validation checks written, its been far easier to write custom functions and control exactly what I want to check and what the output will look like. I've kept the Pandera schemas in here for context but can take them out if you do not want to merge them in. The way they have been created and used is documented in the files in the `notebooks` folder. @hunterowens @KatrinaMKaiser

File: validation_tool/pandera_test.ipynb
Comment (kengie) [reply validation_tool/pandera_test.ipynb:1]: 2 scripts now made. This file was morphed into the 2 more specific ones.

File: validation_tool/schemas/facilities_a15_schema.py (hunk 2 above)
Comment (kengie) [reply validation_tool/schemas/facilities_a15_schema.py]: I deleted this as I pivoted away from the A-15 for now to higher priority things. It was an auto-generated script that can easily be recreated.

File: validation_tool/pandera_test.ipynb
Comment (kengie) [reply validation_tool/pandera_test.ipynb:1]: Added run instructions at top of all scripts and details in notebooks.

**Pull request merged**